pkg/controllers/kube-apiserver.go

```diff
@@ -16,130 +16,193 @@ limitations under the License.
+	s.verbosity = cfg.LogVLevel
+
 	caCertFile := filepath.Join(cfg.DataDir, "certs", "ca-bundle", "ca-bundle.crt")
-	// certDir := filepath.Join(cfg.DataDir, "certs", s.Name())
-	// dataDir := filepath.Join(cfg.DataDir, s.Name())
 
 	if err := s.configureAuditPolicy(cfg); err != nil {
-		klog.Fatalf("Failed to configure kube-apiserver audit policy %v", err)
+		return fmt.Errorf("Failed to configure kube-apiserver audit policy: %w", err)
 	}
 
 	// Get the apiserver port so we can set it as an argument
 	apiServerPort, err := cfg.Cluster.ApiServerPort()
 	if err != nil {
-		// FIXME: This function needs to deal with errors
-		return
+		return err
 	}
 
-	// configure the kube-apiserver instance
-	s.serverOptions = options.NewServerRunOptions()
-
-	s.serverOptions.AllowPrivileged = true
-	s.serverOptions.EnableAggregatorRouting = true
-	s.serverOptions.Features.EnableProfiling = false
-	s.serverOptions.ServiceAccountSigningKeyFile = cfg.DataDir + "/resources/kube-apiserver/secrets/service-account-key/service-account.key"
-	s.serverOptions.ServiceClusterIPRanges = cfg.Cluster.ServiceCIDR
-	s.serverOptions.ServiceNodePortRange = utilnet.PortRange{}
-	// let the type deal with parsing the config file content
-	err = s.serverOptions.ServiceNodePortRange.Set(cfg.Cluster.ServiceNodePortRange)
-	if err != nil {
-		return
+	overrides := &kubecontrolplanev1.KubeAPIServerConfig{
+		APIServerArguments: map[string]kubecontrolplanev1.Arguments{
+			"audit-log-path":    {cfg.AuditLogDir},
+			"audit-policy-file": {cfg.DataDir + "/resources/kube-apiserver-audit-policies/default.yaml"},
+			"client-ca-file":    {caCertFile},
+			"etcd-cafile":       {caCertFile},
+			"etcd-certfile":     {cfg.DataDir + "/resources/kube-apiserver/secrets/etcd-client/tls.crt"},
+			"etcd-keyfile":      {cfg.DataDir + "/resources/kube-apiserver/secrets/etcd-client/tls.key"},
+			"etcd-servers": {
+				"https://127.0.0.1:2379",
+			},
+			"kubelet-certificate-authority": {caCertFile},
+			"kubelet-client-certificate":    {cfg.DataDir + "/resources/kube-apiserver/secrets/kubelet-client/tls.crt"},
+			"kubelet-client-key":            {cfg.DataDir + "/resources/kube-apiserver/secrets/kubelet-client/tls.key"},
+
+			"proxy-client-cert-file":           {cfg.DataDir + "/certs/kube-apiserver/secrets/aggregator-client/tls.crt"},
+			"proxy-client-key-file":            {cfg.DataDir + "/certs/kube-apiserver/secrets/aggregator-client/tls.key"},
+			"requestheader-client-ca-file":     {caCertFile},
+			"service-account-signing-key-file": {cfg.DataDir + "/resources/kube-apiserver/secrets/service-account-key/service-account.key"},
+			"tls-cert-file":                    {cfg.DataDir + "/certs/kube-apiserver/secrets/service-network-serving-certkey/tls.crt"},
+			"tls-private-key-file":             {cfg.DataDir + "/certs/kube-apiserver/secrets/service-network-serving-certkey/tls.key"},
+			"disable-admission-plugins": {
```

Comment:
I see now that the KAS config manifests are just a template used to override values.